app/controllers/spree/admin/base_controller.rb

```diff
@@ -110,6 +111,24 @@ def serializer(ams_prefix)
         name = controller_name.classify
         "::Api::Admin::#{prefix}#{name}Serializer".constantize
       end
+
+      def check_updated_tos_accepted
+        @terms_of_service_banner = false
+
+        return unless spree_user_signed_in?
+
+        return if accepted_tos?
+
+        @terms_of_service_banner = true
+      end
+
+      def accepted_tos?
+        file_uploaded_at = TermsOfServiceFile.updated_at
+
+        current_spree_user.terms_of_service_accepted_at.present? &&
+          current_spree_user.terms_of_service_accepted_at > file_uploaded_at &&
```

Comment:
I think this will work:
```suggestion
        current_spree_user.terms_of_service_accepted_at&.> file_uploaded_at
```
🤔 But is that any more readable...

Reply:
I tried something similar in the console and thought it worked, but oh well, thanks for trying.

Reply:
Another thing: I've just realised that this will cause the banner to suddenly appear for everyone, because `current_spree_user.terms_of_service_accepted_at.present?` will be false to begin with.

I'm not sure if this was already discussed, but I think we should be able to assume that the current terms of service have been accepted. In that case, I suggest we add a migration to set `users.terms_of_service_accepted_at` to the current time (time of deploy). Then people will only be notified _the next time_ the TOS file is updated. What do you think?

Reply:
Sorry, taking a further step back I found that it depends on settings:
![Screen Shot 2023-11-16 at 1 20 10 pm](https://github.com/openfoodfoundation/openfoodnetwork/assets/4188088/9d2d6116-9f87-4628-9c38-90cdc4f49c3f)

So if the Enterprises checkbox isn't checked, they probably haven't confirmed this at signup. So maybe the migration should take that into account.

Should we show the banner at all if this is unchecked?

Reply:
> I'm not sure if this was already discussed, but I think we should be able to assume that the current terms of service have been accepted. In that case, I suggest we add a migration to set `users.terms_of_service_accepted_at` to the current time (time of deploy). Then people will only be notified _the next time_ the TOS file is updated. What do you think?

Thanks for reminding me of this, I meant to check what we should be doing. I raised the question on the original issue as well as the impact of the setting.